control-plane/csi-controller/src/controller.rs

```diff
@@ -532,16 +538,19 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
                 )));
             }
         } else {
-            // Volume is not published, bail out.
-            debug!(
-                "Volume {} is not published, not unpublishing",
-                args.volume_id
-            );
-            return Ok(Response::new(ControllerUnpublishVolumeResponse {}));
+            // Handle situation when the node that hosts nexus is offline and check
+            // volume status explicitly against 'Unknown'.
+            if volume.state.status != VolumeStatus::Unknown {
+                debug!(
+                    "Volume {} is not published, not unpublishing",
+                    args.volume_id
+                );
+                return Ok(Response::new(ControllerUnpublishVolumeResponse {}));
+            }
         }
 
         MayastorApiClient::get_client()
-            .unpublish_volume(&volume_uuid)
+            .unpublish_volume(&volume_uuid, true)
```

Comment:
I think this is probably worth a comment now, stating why we're now saying it's ok to force the unpublish.